src/fu-engine.c

```diff
@@ -2860,16 +2860,26 @@ fu_engine_is_plugin_name_blacklisted (FuEngine *self, const gchar *name)
 static gboolean
 fu_engine_load_plugins (FuEngine *self, GError **error)
 {
+	const gchar *runtime_prefix = NULL;
 	const gchar *fn;
 	g_autoptr(GDir) dir = NULL;
 	g_autoptr(AsProfileTask) ptask = NULL;
+	g_autofree gchar *plugin_directory = NULL;
 
 	/* profile */
 	ptask = as_profile_start_literal (self->profile, "FuEngine:load-plugins");
 	g_assert (ptask != NULL);
 
-	/* search */
-	dir = g_dir_open (PLUGINDIR, 0, error);
+#ifdef SNAP_SUPPORT
+	runtime_prefix = g_getenv ("SNAP");
+	if (runtime_prefix != NULL) {
+		plugin_directory = g_build_filename (runtime_prefix, PLUGINDIR, NULL);
```

Comment:
autofree here pls

Reply:
This one specifically was placed where it was because `plugin_directory` needs to be used later on in the loop.  I was trying to avoid extra `#ifdef` down there, but if you would prefer that i'll swap it around.

Reply:
ahh, no worries then.